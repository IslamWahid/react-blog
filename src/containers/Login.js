import React, { Component } from 'react';
import { FormGroup, FormControl, ControlLabel } from 'react-bootstrap';
import LoaderButton from '../components/LoaderButton';
import axios from 'axios';
import './Login.css';

export default class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      username: ''
    };
  }

  validateForm() {
    return this.state.username.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true });
    try {
      const {
        data: [user]
      } = await axios.get(
        `${process.env.REACT_APP_API_URL}/users?username=${this.state.username}`
      );
      if (!user) throw new Error('username not found');
      this.props.userHasAuthenticated(user);
    } catch (e) {
      this.setState({ isLoading: false });
      alert(e.message);
    }
  };

  render() {
    return (
      <div className="Login">
        <form onSubmit={this.handleSubmit}>
          <FormGroup controlId="username" bsSize="large">
            <ControlLabel>Username</ControlLabel>
            <FormControl
              autoFocus
              type="username"
              value={this.state.username}
              onChange={this.handleChange}
            />
          </FormGroup>
          <LoaderButton
            block
            bsSize="large"
            disabled={!this.validateForm()}
            type="submit"
            isLoading={this.state.isLoading}
            text="Login"
            loadingText="Logging inâ€¦"
          />
        </form>
      </div>
    );
  }
}
