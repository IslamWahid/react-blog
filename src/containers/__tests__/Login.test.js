import React from 'react';
import Login from '../Login';
import { shallow } from 'enzyme';

describe('Login Component', () => {
  let login;

  beforeEach(() => {
    login = shallow(<Login />);
  });
  it('should match snapshot', () => {
    expect(login).toMatchSnapshot();
  });
});
