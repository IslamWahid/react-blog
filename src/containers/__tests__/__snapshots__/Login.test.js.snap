// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <FormGroup
          bsClass="form-group"
          bsSize="large"
          controlId="username"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            autoFocus={true}
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            type="username"
            value=""
          />
        </FormGroup>
        <_default
          block={true}
          bsSize="large"
          disabled={true}
          isLoading={false}
          loadingText="Logging in…"
          text="Login"
          type="submit"
        />
      </form>,
      "className": "Login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="username"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Username
            </ControlLabel>
            <FormControl
              autoFocus={true}
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              type="username"
              value=""
            />
          </FormGroup>,
          <_default
            block={true}
            bsSize="large"
            disabled={true}
            isLoading={false}
            loadingText="Logging in…"
            text="Login"
            type="submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "bsSize": "large",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Username
              </ControlLabel>,
              <FormControl
                autoFocus={true}
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="username"
                value=""
              />,
            ],
            "controlId": "username",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Username",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Username",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "bsClass": "form-control",
                "componentClass": "input",
                "onChange": [Function],
                "type": "username",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": true,
            "bsSize": "large",
            "disabled": true,
            "isLoading": false,
            "loadingText": "Logging in…",
            "text": "Login",
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <FormGroup
            bsClass="form-group"
            bsSize="large"
            controlId="username"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Username
            </ControlLabel>
            <FormControl
              autoFocus={true}
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              type="username"
              value=""
            />
          </FormGroup>
          <_default
            block={true}
            bsSize="large"
            disabled={true}
            isLoading={false}
            loadingText="Logging in…"
            text="Login"
            type="submit"
          />
        </form>,
        "className": "Login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              bsSize="large"
              controlId="username"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Username
              </ControlLabel>
              <FormControl
                autoFocus={true}
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                type="username"
                value=""
              />
            </FormGroup>,
            <_default
              block={true}
              bsSize="large"
              disabled={true}
              isLoading={false}
              loadingText="Logging in…"
              text="Login"
              type="submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "bsSize": "large",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Username
                </ControlLabel>,
                <FormControl
                  autoFocus={true}
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  type="username"
                  value=""
                />,
              ],
              "controlId": "username",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Username",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Username",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": true,
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "onChange": [Function],
                  "type": "username",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "block": true,
              "bsSize": "large",
              "disabled": true,
              "isLoading": false,
              "loadingText": "Logging in…",
              "text": "Login",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
